flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/resources/KubernetesToleration.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.kubeclient.resources;
+
+import io.fabric8.kubernetes.api.model.Toleration;
+import io.fabric8.kubernetes.api.model.TolerationBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Represent Toleration resource in kubernetes.
+ */
+public class KubernetesToleration extends KubernetesResource<Toleration> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(KubernetesToleration.class);
+
+	private KubernetesToleration(Toleration toleration) {
+		super(toleration);
+	}
+
+	public static KubernetesToleration fromMap(Map<String, String> stringMap) {
+		final TolerationBuilder tolerationBuilder = new TolerationBuilder();
+		stringMap.forEach((k, v) -> {
```

Comment:
Could the keys be case insensitive?

Reply:
Thanks for your suggestion. I think it makes sense to ignore character case here.